app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/offline/OfflineDatabaseManagement.kt

```diff
@@ -0,0 +1,194 @@
+        }
+    }
+
+    override suspend fun getRepresentativePicture(categoryId: Any): CategorizedPicture? {
+        require(categoryId is String)
+        return databaseService.getRepresentativePicture(categoryId)
+    }
+
+    override suspend fun putPicture(picture: Uri, category: Category): CategorizedPicture {
+        require(category is OfflineCategory)
+        return try {
+            databaseService.putPicture(picture, category)
+        } catch (e: IllegalArgumentException) {
+            throw e
+        }
+    }
+
+    override suspend fun getCategoryById(categoryId: Any): Category? {
+        require(categoryId is String)
+        return databaseService.getCategory(categoryId)
+    }
+
+    override suspend fun getCategoryByName(categoryName: String): Collection<Category> {
+        val categories = databaseService.getCategories()
+        val res: MutableSet<OfflineCategory> = mutableSetOf()
+        for (c in categories) {
+            if (c.name == categoryName) {
+                res.add(c)
+            }
+        }
+        return res.toSet()
+    }
+
+    override suspend fun putCategory(categoryName: String): Category {
+        return databaseService.putCategory(categoryName)
+    }
+
+    override suspend fun getCategories(): Set<Category> {
+        return databaseService.getCategories()
+    }
+
+    override suspend fun getAllPictures(category: Category): Set<CategorizedPicture> {
+        require(category is OfflineCategory)
+        return try {
+            databaseService.getAllPictures(category)
+        } catch (e: IllegalArgumentException) {
+            throw e
+        }
+    }
+
+    override suspend fun removeCategory(category: Category) {
+        require(category is OfflineCategory)
+        try {
+            databaseService.removeCategory(category)
+        } catch (e: Exception) {
+            //do nothing since this means that the category is not in the database which is the same as having it removed
+        }
+    }
+
+    override suspend fun removePicture(picture: CategorizedPicture) {
+        require(picture is OfflineCategorizedPicture)
+        try {
+            databaseService.removePicture(picture)
+        } catch (e: Exception) {
+            //do nothing since this means that the picture is not in the database which is the same as having it removed
+        }
+    }
+
+    override suspend fun putDataset(name: String, categories: Set<Category>): Dataset {
+        return databaseService.putDataset(name, categories)
+    }
+
+    override suspend fun getDatasetById(id: Any): Dataset? {
+        require(id is String)
+        return databaseService.getDataset(id)
+    }
+
+    override suspend fun getDatasetByName(datasetName: String): Collection<Dataset> {
+        val datasets = databaseService.getDatasets()
+        val res: MutableSet<OfflineDataset> = mutableSetOf()
+        for (d in datasets) {
+            if (d.name == datasetName) {
+                res.add(d as OfflineDataset)
```

Comment:
```suggestion
                res.add(d)
```
superflous (per compiler warnings)